packages/core/src/components/spinner/spinner.tsx

```diff
@@ -76,81 +76,21 @@ export interface SpinnerProps<T extends HTMLElement = HTMLElement> extends Props
 
-    public componentDidUpdate(prevProps: SpinnerProps) {
-        if (prevProps.value !== this.props.value) {
-            // IE/Edge: re-render after changing value to force SVG update
-            this.forceUpdate();
-        }
-    }
-
-    public render() {
-        const { className, intent, value, tagName = "div", ...htmlProps } = this.props;
-        const size = this.getSize();
-
-        const classes = classNames(
-            Classes.SPINNER,
-            Classes.intentClass(intent),
-            { [Classes.SPINNER_NO_SPIN]: value != null },
-            className,
-        );
-
-        // keep spinner track width consistent at all sizes (down to about 10px).
-        const strokeWidth = Math.min(MIN_STROKE_WIDTH, (STROKE_WIDTH * SpinnerSize.LARGE) / size);
-        const strokeOffset = PATH_LENGTH - PATH_LENGTH * (value == null ? 0.25 : clamp(value, 0, 1));
-
-        // multiple DOM elements around SVG are necessary to properly isolate animation:
-        // - SVG elements in IE do not support anim/trans so they must be set on a parent HTML element.
-        // - SPINNER_ANIMATION isolates svg from parent display and is always centered inside root element.
-        return React.createElement(
-            tagName,
-            {
-                "aria-label": "loading",
-                "aria-valuemax": 100,
-                "aria-valuemin": 0,
-                "aria-valuenow": value === undefined ? undefined : value * 100,
-                className: classes,
-                role: "progressbar",
-                ...htmlProps,
-            },
-            React.createElement(
-                tagName,
-                { className: Classes.SPINNER_ANIMATION },
-                <svg
-                    width={size}
-                    height={size}
-                    strokeWidth={strokeWidth.toFixed(2)}
-                    viewBox={this.getViewBox(strokeWidth)}
-                >
-                    <path className={Classes.SPINNER_TRACK} d={SPINNER_TRACK} />
-                    <path
-                        className={Classes.SPINNER_HEAD}
-                        d={SPINNER_TRACK}
-                        pathLength={PATH_LENGTH}
-                        strokeDasharray={`${PATH_LENGTH} ${PATH_LENGTH}`}
-                        strokeDashoffset={strokeOffset}
-                    />
-                </svg>,
-            ),
-        );
-    }
-
-    protected validateProps({ className = "", size }: SpinnerProps) {
+    React.useEffect(() => {
         if (size != null && (className.indexOf(Classes.SMALL) >= 0 || className.indexOf(Classes.LARGE) >= 0)) {
             console.warn(SPINNER_WARN_CLASSES_SIZE);
         }
-    }
+    }, [className, size]);
 
     /**
      * Resolve size to a pixel value.
      * Size can be set by className, props, default, or minimum constant.
      */
-    private getSize() {
-        const { className = "", size } = this.props;
+    const sizePx = React.useMemo(() => {
```

Comment:
This component isn't memoized and this doesn't look expensive, I think we should remove the `useMemo` here and either extract this as a function below the component or make this an IIFE

Reply:
d04cc1da26e00cd49f4b23e72df64dadc94eb7df